src/group_theory/index.lean

```diff
@@ -132,6 +132,38 @@ begin
   exact relindex_mul_relindex (H ⊓ L) (K ⊓ L) L (inf_le_inf_right L hHK) inf_le_right,
 end
 
+/-- A subgroup has index two if and only if there exists `a` such that for all `b`, exactly one
+of `b * a` and `b` belong to `H`. -/
+@[to_additive "/-- An additive subgroup has index two if and only if there exists `a` such that for
+all `b`, exactly one of `b + a` and `b` belong to `H`. -/"]
+lemma index_eq_two_iff : H.index = 2 ↔ ∃ a, ∀ b, b * a ∈ H ↔ b ∉ H :=
```

Comment:
Have you considered using `xor` to express the "exactly one of"?

Reply:
Done, also added lemmas about `xor`.